File: app_codebase/python_base_05/bin/modules/ws/demo_ws_app.py
"""Register demo WebSocket channels on all tiers."""

from core.state.room.room_demo_handler import handle_demo_room_message
from core.ws.contracts.register_channel_contract import ApplicationChannelSink
from modules.ws.demo_ws_service import handle_demo_echo, handle_system


def register_demo_ws_channels(channels: ApplicationChannelSink) -> None:
    for register in (
        channels.public_channel,
        channels.authuser_channel,
        channels.service_channel,
    ):
        register("system", handle_system)
        register("demo/echo", handle_demo_echo)
        register("demo/room", handle_demo_room_message)

File: app_codebase/dart_bkend_base_02/bin/modules/ws/demo_ws_app.dart
/// Register demo WebSocket channels on all tiers.
library;

import '../../core/ws/contracts/register_channel_contract.dart';
import 'demo_ws_service.dart';

void registerDemoWsChannels(ApplicationChannelSink channels) {
  for (final register in [
    channels.publicChannel,
    channels.authuserChannel,
    channels.serviceChannel,
  ]) {
    register('system', handleSystem);
    register('demo/echo', handleDemoEcho);
    register('demo/room', handleDemoRoom);
  }
}
